packages/loader/container-loader/src/container.ts

```diff
@@ -1733,13 +1728,53 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
         // Allow the protocol handler to process the message
         const result = this.protocolHandler.processMessage(message, local);
-        this.collabWindowTracker.scheduleSequenceNumberUpdate(message, result.immediateNoOp === true);
+        // Inactive (not in quorum or not writers) clients don't take part in the minimum sequence number calculation.
+        if (this.activeConnection()) {
+            if (this.collabWindowTracker === undefined) {
+                // Note that config from first connection will be used for this container's lifetime.
+                // That means that if relay service changes settings, such changes will impact only newly booted
+                // clients.
+                // All existing will continue to use settings they got earlier.
+                const [noopTimeFrequency, noopCountFrequency] = this.getNoopConfig();
+                this.collabWindowTracker = new CollabWindowTracker(
+                    (type, contents) => {
+                        assert(this.activeConnection(),
+                            0x241 /* "disconnect should result in stopSequenceNumberUpdate() call" */);
+                        this.submitMessage(type, contents);
+                    },
+                    noopTimeFrequency,
+                    noopCountFrequency,
+                );
+            }
+            this.collabWindowTracker.scheduleSequenceNumberUpdate(message, result.immediateNoOp === true);
+        }
 
         this.emit("op", message);
 
         return result;
     }
 
+    /**
+     *  #260 (ADO)
+     * back-compat: noopTimeFrequency & noopCountFrequency properties were added to
+     * IClientConfiguration in 0.59.3000. During the integration, we must read the
+     * available configuration from the loader options.
+     */
+    private getNoopConfig(): [number | undefined, number | undefined] {
+        const serviceConfiguration = this.serviceConfiguration as any;
```

Comment:
> as any [](http://example.com/codeflow?start=62&length=7)

Just so I can understand why do we need the cast as any  here ?

Reply:
In this case, the object may or may not have the properties. When it doesn't, the compiler will complain. Currently, it will not have them, as the package needs a version bump from the definitions, where the properties were added. Afterwards, we can remove the case altogether. This is one of those (back|forward)-compat hacks which we sometimes do in order to deal with staged changes.

Reply:
Thanks - I've just realized the previous version did not have these 2 properties ;)